### DISCUSSION TITLE
large integer as property triggers "corrupt data when trying to hydrate the component"

### DISCUSSION THREAD

**Discussion by gorghoa** [livewire/livewire#4788] [7. Bugs (archived)]
Hi,  

First of all, thank you for livewire, it really makes our life easier :heart: !

We recently were bitten by a "Livewire encountered corrupt data when trying to hydrate the … component" exception when using a component (livewire 2.10.4).

We eventually found out that one of our property was a very large integer, larger than the javascript `Number.MAX_SAFE_INTEGER` ([mdn](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/MAX_SAFE_INTEGER)) which can be lower than the php max limit `PHP_INT_MAX` ([php doc](https://www.php.net/manual/en/reserved.constants.php)).

The value of the js integer  was not the same as the php one => hence corruption.

herafter, a crash test:

```php
class Corruption extends Livewire\Component\Component
{
    public int $myInt;

    public function mount() {
        $this->myInt = PHP_INT_MAX;
    }

    public function render(): mixed
    {
        return <<<'blade'
            <div>
              {{$myInt}}

              <button wire:click="corrupt">corrupt</button>
            </div>
        blade;
    }

    public function corrupt(): never
    {
       throw new \Exception("Should not be called, a corrupt error should have occured");
    }
}
```

**Comment by joshhanley** [comment]
@gorghoa yeah that is because you've run into javascript's integer limit.

So if you need really big ints, you should convert them to strings before they get sent to the browser.

Hope this helps!

**Reply by gorghoa** [reply]
Hi @joshhanley, thank you for taking time answering me! 

> @gorghoa yeah that is because you've run into javascript's integer limit.

Yes, that’s what I say in the report ;)

I’m not looking for solution (I’ve already added a comment on [this stackoverfow discussion](https://stackoverflow.com/questions/63848826/livewire-encountered-corrupt-data-when-trying-to-hydrate-the-component/71427329#71427329) if it can help people), but rather wanted to fill a bug report here.

cheers!

**Comment by fcno** [comment]
Same problem here!
